server/server.go

```diff
@@ -113,14 +114,18 @@ func (keys *DefaultKeystore) AddFromFile(path string, LoadKey func([]byte) (cryp
 // AddFromURI loads all keys matching the given PKCS#11 URI to the keystore. LoadURI
 // is called to parse the URL, connect to the module, and populate a crypto.Signer,
 // which is stored in the Keystore.
-func (keys *DefaultKeystore) AddFromURI(uri string, LoadURI func(string) (crypto.Signer, error)) error {
+func (keys *DefaultKeystore) AddFromURI(uri string, LoadPKCS11URI func(string) (crypto.Signer, error)) error {
 	log.Infof("loading %s...", uri)
-
-	priv, err := LoadURI(uri)
+	var priv crypto.Signer
+	var err error
+	if azure.IsKeyVaultURI(uri) {
```

Comment:
PKCS11 has a scheme registered in RFC 7512 https://tools.ietf.org/html/rfc7512#page-4. Now that we're generalizing this, I think it makes sense to check for that scheme ahead of calling the LoadPKCS11URI, for better uniformity of errors, and it makes more sense once we add a third option (which by the one, none, or many rule, we will).

Reply:
oh awesome, I wasn't sure if the docs indicating `pkcs:` was an accident or not :)

Reply:
hah turns out there's already a regex in the code for it, I moved it so it's not behind the buildflags in 84aadce03bc68cd82e4aac686cb6764f5f6877dd